rls/src/main/java/io/grpc/rls/RlsProtoConverters.java

```diff
@@ -155,25 +177,30 @@ private static Long convertTimeIfNotNull(TimeUnit from, TimeUnit to, Long value)
 
     @SuppressWarnings("unchecked")
     static GrpcKeyBuilder convert(Map<String, ?> keyBuilder) {
-      List<Map<String, ?>> rawNames =
-          JsonUtil.checkObjectList(JsonUtil.getList(keyBuilder, "names"));
+      List<?> rawRawNames = JsonUtil.getList(keyBuilder, "names");
+      checkArgument(
+          rawRawNames != null && !rawRawNames.isEmpty(),
+          "each keyBuilder must have at least one name");
+      List<Map<String, ?>> rawNames = JsonUtil.checkObjectList(rawRawNames);
       List<Name> names = new ArrayList<>();
       for (Map<String, ?> rawName : rawNames) {
-        names.add(
-            new Name(
-                JsonUtil.getString(rawName, "service"), JsonUtil.getString(rawName, "method")));
+        String serviceName = JsonUtil.getString(rawName, "service");
+        checkArgument(!Strings.isNullOrEmpty(serviceName), "service must not be empty or null");
+        names.add(new Name(serviceName, JsonUtil.getString(rawName, "method")));
       }
+      List<?> rawRawHeaders = JsonUtil.getList(keyBuilder, "headers");
       List<Map<String, ?>> rawHeaders =
-          JsonUtil.checkObjectList(JsonUtil.getList(keyBuilder, "headers"));
+          rawRawHeaders == null
+              ? new ArrayList<Map<String, ?>>()
+              : JsonUtil.checkObjectList(JsonUtil.getList(keyBuilder, "headers"));
```

Comment:
I think here the var can be reused, same as previously:
```suggestion
              : JsonUtil.checkObjectList(rawRawHeaders);
```

Reply:
Done.